wolfcrypt/src/port/xilinx/xil-aesgcm.c

```diff
@@ -44,14 +42,272 @@
     #include <wolfcrypt/src/misc.c>
 #endif
 
-#include "xparameters.h"
-
 enum {
     AEAD_NONCE_SZ       = 12,
     AES_GCM_AUTH_SZ     = 16, /* AES-GCM Auth Tag length    */
 };
 
 
+#ifdef HAVE_AESGCM
+
+#if defined(WOLFSSL_XILINX_CRYPT_VERSAL)
+
+#include "xsecure_aesclient.h"
+
+typedef int (*init_fn)(XSecure_ClientInstance *InstancePtr, XSecure_AesKeySource KeySrc, u32 Size, u64 IvAddr);
+
+enum aad_state {
+    SW_AAD,
+    HW_ENGINE_AAD
+};
+
+static int check_keysize(word32 len)
+{
+    if (len != 16u && len != 32u) {
```

Comment:
Lets use AES_128_KEY_SIZE and AES_256_KEY_SIZE enum from wolfssl/wolfcrypt/aes.h instead of the hard set values.